Realm.Shared/RealmObject.cs

```diff
@@ -64,27 +64,30 @@ internal void _CopyDataFromBackingFieldsToRow()
             Debug.Assert(this.IsManaged);
 
             var thisType = this.GetType();
-            var wovenProperties = from prop in thisType.GetProperties()
+            var wovenPropertiesWithBacking = from prop in thisType.GetProperties()
                                   let backingField = prop.GetCustomAttributes(false)
                                                          .OfType<WovenPropertyAttribute>()
                                                          .Select(a => a.BackingFieldName)
                                                          .SingleOrDefault()
                                   where backingField != null
                                   select new { Info = prop, Field = thisType.GetField(backingField, BindingFlags.Instance | BindingFlags.NonPublic) };
 
-            foreach (var prop in wovenProperties)
+            foreach (var prop in wovenPropertiesWithBacking)
             {
                 var value = prop.Field.GetValue(this);
-                if (prop.Info.PropertyType.IsGenericType)
-                {
-                    var genericType = prop.Info.PropertyType.GetGenericTypeDefinition();
-                    if (genericType == typeof(RealmList<>))
+                if (value != null) {
+                    var listValue = value as IEnumerable<RealmObject>;
+                    if (listValue != null)  // assume it is NOT a RealmList so need to wipe afer copy
```

Comment:
Ah, apologies, I mistakenly though this was rebased against the Schema API changed in master.